src/Utilities/Compiler/CodeMetrics/MetricsHelper.cs

```diff
@@ -86,11 +86,42 @@ internal static async Task<long> GetLinesOfCodeAsync(ImmutableArray<SyntaxRefere
                     // Declaration on a single line, we count it as a separate line.
                     delta = 1;
                 }
+                else
+                {
+                    // Ensure that we do not count the leading and trailing emtpy new lines.
+                    var additionalNewLines = Math.Max(0, GetNewlineCount(declSyntax.GetLeadingTrivia(), leading: true) + GetNewlineCount(declSyntax.GetTrailingTrivia(), leading: false) - 1);
+                    delta -= additionalNewLines;
+                }
 
                 linesOfCode += delta;
             }
 
             return linesOfCode;
+
+            static int GetNewlineCount(SyntaxTriviaList trivialList, bool leading)
+            {
+                var triviaParts = trivialList.ToFullString().Split(new[] { "\r\n", "\r", "\n" }, StringSplitOptions.None).ToImmutableArray();
+                return GetNewlineCount(triviaParts, leading);
+
+                static int GetNewlineCount(ImmutableArray<string> triviaParts, bool leading)
+                {
+                    var index = leading ? 0 : triviaParts.Length - 1;
+                    var loopCondition = leading ? LoopConditionForLeading : (Func<int, int, bool>)LoopConditionForTrailing;
+                    var incrementOrDecrement = leading ? 1 : -1;
+                    var count = 0;
+                    for (; loopCondition(index, triviaParts.Length); index += incrementOrDecrement, count++)
+                    {
+                        var trivia = triviaParts[index].Trim();
+                        if (trivia.Length != 0)
```

Comment:
>Length [](start = 35, length = 6)

Do comments count as syntax trivia? Didn't see any unit tests with comments in the test code.

Reply:
Yes, good point, let me add a test.